File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/messages-module.ts
import { buildTextEditRequest } from './builders.js';
import type {
  EditMessageRequest,
  EditTextMessageOptions,
  MessageMutationResult,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatMessagesModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  edit(
    messageId: string | number,
    body: EditMessageRequest,
  ): Promise<MessageMutationResult> {
    return this.context.backendClient.message.edit(messageId, body);
  }

  editText(
    messageId: string | number,
    text: string,
    options: EditTextMessageOptions = {},
  ): Promise<MessageMutationResult> {
    return this.edit(messageId, buildTextEditRequest(text, options));
  }

  recall(messageId: string | number): Promise<MessageMutationResult> {
    return this.context.backendClient.message.recall(messageId);
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/messages_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'builders.dart';
import 'context.dart';
import 'types.dart';

class CrawChatMessagesModule {
  final CrawChatSdkContext context;

  CrawChatMessagesModule(this.context);

  Future<MessageMutationResult?> edit(
    String messageId,
    EditMessageRequest body,
  ) {
    return context.backendClient.message.edit(messageId, body);
  }

  Future<MessageMutationResult?> editText(
    String messageId, {
    required String text,
    CrawChatTextEditOptions options = const CrawChatTextEditOptions(),
  }) {
    return edit(
      messageId,
      CrawChatBuilders.textEdit(text: text, options: options),
    );
  }

  Future<MessageMutationResult?> recall(String messageId) {
    return context.backendClient.message.recall(messageId);
  }
}
